src/renderer/services/bitcoin/ledger.ts

```diff
@@ -1,20 +1,30 @@
 import * as RD from '@devexperts/remote-data-ts'
+import * as E from 'fp-ts/Either'
 import * as FP from 'fp-ts/lib/function'
 import * as Rx from 'rxjs'
-import { catchError, map, startWith } from 'rxjs/operators'
+import { catchError, map, startWith, switchMap } from 'rxjs/operators'
 
+import { LedgerErrorId } from '../../../shared/api/types'
 import { observableState } from '../../helpers/stateHelper'
 import { LedgerAddressRD } from '../wallet/types'
 import { LedgerService } from './types'
 
 const { get$: ledgerAddress$, set: setLedgerAddressRD } = observableState<LedgerAddressRD>(RD.initial)
 
+const errorHandler = (value: E.Either<LedgerErrorId, string>) => {
+  if (E.isLeft(value)) {
+    return Promise.reject(value.left)
+  }
+  return Promise.resolve(value.right)
+}
+
```

Comment:
No need to transform `E.Either<LedgerErrorId, string>` -> `Promise<string>` (see next comment why)